Mathlib/Algebra/Module/LinearMap.lean

```diff
@@ -86,10 +86,11 @@ is semilinear if it satisfies the two properties `f (x + y) = f x + f y` and
 maps is available with the predicate `IsLinearMap`, but it should be avoided most of the time. -/
 structure LinearMap {R S : Type*} [Semiring R] [Semiring S] (σ : R →+* S) (M : Type*)
     (M₂ : Type*) [AddCommMonoid M] [AddCommMonoid M₂] [Module R M] [Module S M₂] extends
-    AddHom M M₂ where
-  /-- A linear map preserves scalar multiplication.
-  We prefer the spelling `_root_.map_smul` instead. -/
-  protected map_smul' : ∀ (r : R) (x : M), toFun (r • x) = σ r • toFun x
+    AddHom M M₂, MulActionHom σ M M₂
```

Comment:
(Semi-)LinearMap appears to be exactly the same as DistribMulActionSemiHom, except with different typeclass prerequisites (which allow map_zero to be derived rather than included). I understand that replacing LinearMap with DistribMulActionSemiHom requires introducing another constructor that doesn't require providing map_zero, and switching from `where` to the new constructor, but I think SemilinearMapClass can be replaced by DistribMulActionSemiHomClass.

By the way, why do you add `Semi` to `DistribMulActionHom` but keep the name `MulActionHom` unchanged?

Reply:
Thank you for these comments. You point out rightly that the naming is still a mess. I will write something on Zulip to clarify which names should be given to these classes.

Reply:
@alreadydone 
I'm slowly updating the files to take your last remark into account.
Also, when the map between acting types is identity, I provide the richest possible identity (eg, `MonoidHom.id M` instead of `@id M`).

Reply:
Regarding your first remark, I have the impression to `LinearMap` and `DistribMulActionHom` are identical up to instances regarding the relevant types. 
I am tempted to try *renaming* `DistribMulActionHom` as `LinearMap`…